validator/delphi_validator/validate.py

```diff
@@ -1,237 +1,721 @@
+        export_files = read_filenames(export_dir)
+        date_filter = make_date_filter(self.start_date, self.end_date)
+
+        # Make list of tuples of CSV names and regex match objects.
+        validate_files = [(f, m) for (f, m) in export_files if date_filter(m)]
+
+        # Get all expected combinations of geo_type and signal.
+        geo_sig_cmbo = get_geo_sig_cmbo(self.data_source)
+
+        self.check_missing_dates(validate_files)
+        self.check_settings()
+
+        all_frames = []
+
+        # Individual file checks
+        # For every daily file, read in and do some basic format and value checks.
+        for filename, match in validate_files:
+            data_df = load_csv(join(export_dir, filename))
+
+            self.check_df_format(data_df, filename)
+            self.check_bad_geo_id(
+                data_df, filename, match.groupdict()['geo_type'])
+            self.check_bad_val(data_df, filename, match.groupdict()['signal'])
+            self.check_bad_se(data_df, filename)
+            self.check_bad_sample_size(data_df, filename)
+
+            # Get geo_type, date, and signal name as specified by CSV name.
+            data_df['geo_type'] = match.groupdict()['geo_type']
+            data_df['date'] = match.groupdict()['date']
+            data_df['signal'] = match.groupdict()['signal']
+
+            # Add current CSV data to all_frames.
+            all_frames.append(data_df)
+
+        all_frames = pd.concat(all_frames)
+
+        # Get list of dates we expect to see in all the CSV data.
+        date_slist = all_frames['date'].unique().tolist()
+        date_list = list(
+            map(lambda x: datetime.strptime(x, '%Y%m%d'), date_slist))
+
+        # recent_lookbehind: start from the check date and working backward in time,
+        # how many days do we want to check for anomalies?
+        # Choosing 1 day checks just the daily data.
+        recent_lookbehind = timedelta(days=1)
+
+        # semirecent_lookbehind: starting from the check date and working backward
+        # in time, how many days do we use to form the reference statistics.
+        semirecent_lookbehind = timedelta(days=7)
+
+        smooth_option_regex = re.compile(r'([^_]+)')
+
+        # Keeps script from checking all files in a test run.
+        if self.test_mode:
+            kroc = 0
+
+        # Comparison checks
+        # Run checks for recent dates in each geo-sig combo vs semirecent (last week) API data.
+        for geo_sig_df, geo, sig in read_geo_sig_cmbo_files(
+                geo_sig_cmbo,
+                export_dir,
+                [name_match_pair[0] for name_match_pair in validate_files],
+                date_slist):
+
+            match_obj = smooth_option_regex.match(sig)
+            smooth_option = match_obj.group(1)
+
+            if smooth_option not in ('raw', 'smoothed'):
+                smooth_option = 'smoothed' if '7dav' in sig or 'smoothed' in sig else 'raw'
```

Comment:
This is likely to be brittle -- need to keep in mind when naming future signals that the validator will assume raw unless one of these keywords is present. Probably fine for now but we may want to make the smoothed keyword and/or default behavior configurable some time in the future.

Reply:
I'll add this and the new "lag_option" (or similar name) as user settings.

Reply:
User can now specify a list of `smoothed_signals` in the params file. If a signal name appears in the list, the lower threshold is used when comparing test and reference (API) avg values; otherwise, the signal is assumed to be "raw" and the higher threshold is used.